File: MyLeetCode/python/ Valid Word Square.py
###
# Array
# Time Complexity: O(m*n)
# Space Complexity: O(1)
###
class Solution(object):
    def validWordSquare(self, words):
        """
        :type words: List[str]
        :rtype: bool
        """
        
        for i, word in enumerate(words):
            for j, c in enumerate(word):
                if j >= len(words) or i >= len(words[j]) or c != words[j][i]:
                    return False
        return True
        

File: MyLeetCode/Leetcode/Valid Word Square.cpp
class Solution {
public:
    bool validWordSquare(vector<string>& words) {
        int rows = words.size();
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < words[i].size(); j++) {
                if (j >= rows || i >= words[j].size() || words[i][j] != words[j][i]) return false;
            }
        }
        return true;
    }
};